src/OpenBinary/reader.jl

```diff
@@ -133,18 +125,17 @@ function PSRI.open(
         _first_stage = read(ioh, Int32) #month or week
         first_year = read(ioh, Int32) # year
         #
-        unit_buffer = Vector{Cchar}(undef, 7)
-        read!(ioh, unit_buffer)
-        unit_str = strip(join(unit_buffer))
+        unit_str = strip(join(Char.(read!(ioh, Vector{Cchar}(undef, 7)))))
```

Comment:
tricky to read this one. IMO this should be either separated or written with pipes to be more clear `read!(ioh, Vector{Cchar}(undef, 7)) .|> Char |> join |> strip`